docs/devguide/introduction.rst

```diff
@@ -276,6 +276,70 @@ OpenDDS supports the following building blocks, with notes/caveats listed below
+*********************************************
+
+The two components needed for interoperability are the :ref:`RTPS Discovery library <rtps-disc>` and the :ref:`RTPS UDP transport library <rtps-udp-transport>`.
+Both of these libraries are built and installed by default.
+Those using a custom build system need to make sure that these libraries are available to the application.
+
+As of OpenDDS 3, these components will not be used by default and must be configured.
+This can be done via :ref:`environment variables, command-line options, configuration files, and/or code <config-store-keys>`.
+
+The following configuration file snippet changes the default discovery to RTPS and the default transport to RTPS UDP.
+
+.. code-block:: ini
+
+    [common]
+    DCPSDefaultDiscovery=DEFAULT_RTPS
+    DCPSGlobalTransportConfig=$file
+
+    [transport/the_rtps_transport]
+    transport_type=rtps_udp
+
+The following code accomplishes the same:
+
+.. code-block:: cpp
+
+    using namespace OpenDDS::DCPS;
+    using namespace OpenDDS::RTPS;
+    TransportConfig_rch config =
+      TransportRegistry::instance()->create_config("rtps_interop_demo");
+    TransportInst_rch inst =
+      TransportRegistry::instance()->create_inst("rtps_transport","rtps_udp");
+    config->instances_.push_back(inst);
+    TransportRegistry::instance()->global_config(config);
+
+    RtpsDiscovery_rch disc = make_rch<RtpsDiscovery>("RtpsDiscovery");
+    disc->use_xtypes(RtpsDiscoveryConfig::XTYPES_NONE);
+    TheServiceParticipant->add_discovery(static_rchandle_cast<Discovery>(disc));
+    TheServiceParticipant->set_default_discovery(disc->key());
+
+The remaining concern for interoperability is the use of :ref:`xtypes` features.
+See :ref:`xtypes--unimplemented-features` and :ref:`xtypes--differences-from-the-specification`.
+If XTypes is not being used, then see :ref:`xtypes--interoperability-with-non-xtypes-implementations`.
+
+For maximum interoperability, complete type objects should be used.
+See :ref:`xtypes--enabling-use-of-completetypeobjects`.
+
+:ref:`opendds_idl` has options for changing the defaults when processing types.
+Using these options may reduce the number of annotations needed in IDL.
+See :option:`opendds_idl --default-nested`, :option:`opendds_idl --default-extensibility`, :option:`opendds_idl --default-autoid`, and :option:`opendds_idl --default-try-construct`.
+
+For maximum interoperability, the :ref:`xtypes--xcdr2` data representation should be used.
+OpenDDS has limited support for XCDR1 (see :ref:`xtypes--xcdr1-support`).
+XCDR2 is the default.
+It can be set explicitly via QoS or by using the non-standard :ref:`xtypes--anno-opendds-data-representation-xcdr2` annotation.
+The following snippet shows how to choose XCDR2 for a DataWriter:
+
+.. code-block:: cpp
+
+    DDS::DataWriterQos qos;
+    pub->get_default_datawriter_qos(qos);
+    qos.representation.value.length(1);
+    qos.representation.value[0] = DDS::XCDR2_DATA_REPRESENTATION;
```

Comment:
This seems kinda redundant with this in the xtypes chapter:

https://github.com/OpenDDS/OpenDDS/blob/9bcc5b2b6e2fbdfb5d445b2e1be168d948d53477/docs/devguide/xtypes.rst?plain=1#L451-L458